src/api-collector/main.go

```diff
@@ -0,0 +1,218 @@
+}
+
+func getOrgRepos(ctx context.Context, gitOwner string, client *github.Client) ([]*github.Repository, error) {
+	opt := &github.RepositoryListByOrgOptions{
+		ListOptions: github.ListOptions{},
+	}
+
+	var allRepos []*github.Repository
+
+	for {
+		repos, response, err := client.Repositories.ListByOrg(ctx, gitOwner, opt)
+		if err != nil {
+			return nil, err
+		}
+		allRepos = append(allRepos, repos...)
+		if response.NextPage == 0 {
+			break
+		}
+		opt.Page = response.NextPage
+	}
+
+	return allRepos, nil
+}
+
+func getAPISpecAssets(ctx context.Context, client *github.Client, owner string, repo string) []assetInfo {
+	var apiSpecs []assetInfo
+	release, _, err := client.Repositories.GetLatestRelease(ctx, owner, repo)
+	if err != nil {
+		log.Println("\t- No release found")
+		return apiSpecs
+	}
+	log.Printf("\t+ Latest release found: %s\n", *release.Name)
+	assets, _, err := client.Repositories.ListReleaseAssets(ctx, owner, repo, *release.ID, nil)
+	if err != nil {
+		log.Println("\t- No assets found in the release")
+		return apiSpecs
+	}
+	for _, asset := range assets {
+		if strings.Contains(*asset.Name, "openapi.yaml") {
+			apiSpecs = append(apiSpecs, assetInfo{*asset.ID, *asset.Name})
+		}
+	}
+	return apiSpecs
+}
+
+func downloadAPISpecs(ctx context.Context, client *github.Client, owner string, repo string, assets []assetInfo) []string {
+	var downloadedSpecs []string
+	for _, asset := range assets {
+		assetReader, assetURL, err := client.Repositories.DownloadReleaseAsset(ctx, owner, repo, asset.id, nil)
+		if err != nil {
+			log.Printf("\t- Error downloading OpenAPI spec %s: %s\n", asset.name, err)
+			continue
+		}
+		var reader io.ReadCloser
+		if assetReader == nil {
+			resp, err := http.Get(assetURL)
+			if err != nil {
+				log.Printf("\t- Error downloading OpenAPI spec %s: %s\n", asset.name, err)
+				continue
+			}
+			reader = resp.Body
+			defer resp.Body.Close()
+		} else {
+			reader = assetReader
+			defer assetReader.Close()
+		}
+		content, err := io.ReadAll(reader)
+		if err != nil {
+			log.Printf("\t- Error reading OpenAPI spec %s: %s\n", asset.name, err)
+			continue
+		}
+		var spec OpenAPISpec
+		err = yaml.Unmarshal(content, &spec)
+		if err != nil {
+			log.Printf("\t- Error parsing OpenAPI spec yaml format: %s\n", err)
+			continue
+		}
+		dirPath := path.Join("docs", repo, spec.Info.Version)
+		err = os.MkdirAll(dirPath, os.ModePerm)
+		if err != nil {
+			log.Printf("\t- Error creating directory: %s\n", err)
+			continue
+		}
+		filePath := path.Join(dirPath, asset.name)
+		err = os.WriteFile(filePath, content, 0644)
+		if err != nil {
+			log.Printf("\t- Error saving OpenAPI spec content to file: %s\n", err)
+			continue
+		}
+		downloadedSpecs = append(downloadedSpecs, filePath)
+		log.Printf("\t+ OpenAPI spec %s downloaded successfully\n", asset.name)
+
+	}
+	return downloadedSpecs
+}
+
+func uploadAPISpecs(ctx context.Context, client *github.Client, owner string, repo string, specs []string) {
```

Comment:
Is this actually an "upload", or is it doing a git commit? In this case, I would also call it something like `commitDownloadedSpec`

Reply:
https://github.com/eclipse-tractusx/api-hub/pull/1/commits/ada7aaecca5abfcd6a85d80820e5709904950344